### ISSUE TITLE
Expression syntax iif doesn't work

### ISSUE THREAD

**Issue by baio** [angular/angular#5178]
```
[value]="selected ? selected.label : null"
```

doen't wotk 

EXCEPTION: TypeError: Cannot read property 'label' of null in [null]

**Comment by ericmartinezr** [comment]
What's that? An object of yours? Does your object contain a property called `label`? Did you try using the elvis operator `[value]="selected ? selected?.label : null"`?

**Comment by IgorMinar** [comment]
Are you trying to use a ternary operator? Or elvis operator?

**Comment by baio** [comment]
Elvis operator works

```
selected ? selected?.label : null
```

Is it intended behavior? Have feeling that elvis not necessary in this case.

**Comment by pkozlowski-opensource** [comment]
> Have feeling that elvis not necessary in this case.

You are right, it shouldn't be necessary in this case. But there was a bug #4790 that prevented it from working correctly. It is already fixed in master (7e92d2e6b7e77fb019a147e4728c034968d82bdb) but not released yet - will be part of the next release.

Closing as this is already fixed in master.
